client/src/app/components/stock-search/stock-search.component.html

```diff
@@ -12,14 +12,51 @@
   #auto="matAutocomplete"
   (optionActivated)="getActiveOption($event)"
 >
-  <mat-option
-    *ngFor="let stock of searchResults"
-    [value]="stock"
-    [routerLink]="'/stock-detail/' + stock.Symbol"
-    (click)="query=stock.Name"
-  >
-    <div class="option-container">
-      <span>{{ stock.Name }}</span> <span>{{ stock.Symbol }}</span>
+  <div class="searchOptions">
+    <span>Filter:</span>
+    <div
+      (click)="selectedSearchOption = 'all'; queryFilter(null)"
+      [class]="isSelectedClass('all')"
+    >
+      <mat-icon>select_all</mat-icon><span>All</span>
     </div>
-  </mat-option>
+    <div
+      (click)="selectedSearchOption = 'stocks'; queryFilter(null)"
+      [class]="isSelectedClass('stocks')"
+    >
+      <mat-icon>trending_up</mat-icon>
+      <span>Stocks</span>
+    </div>
+    <div
+      (click)="selectedSearchOption = 'users'; queryFilter(null)"
+      [class]="isSelectedClass('users')"
+    >
+      <mat-icon>people_outline</mat-icon>
+      <span>Users</span>
+    </div>
+  </div>
+  <mat-divider></mat-divider>
+  <div class="custom-scroll">
+    <mat-option
+      *ngFor="let item of searchResults"
+      [value]="item"
+      [routerLink]="redirectSearch(item.type) + item.id"
+      (click)="query = item.name"
```

Comment:
Not sure if it worth having a function for this click, I like this one too!

Reply:
Its needed because i have to identify if it is a stock or a user